src/Microsoft.AspNetCore.Authentication.MicrosoftAccount/MicrosoftAccountHelper.cs

```diff
@@ -34,8 +35,17 @@ public static string GetName(JObject user)
             {
                 throw new ArgumentNullException(nameof(user));
             }
+            
+            //displayName is required for Azure AD accounts, but not for Microsoft accounts.
+            //givenName & surename is required for Microsoft accounts, but not for Azure AD accounts
+            //In case of microsoft accounts displayName is null => fallback to givenName + surename
+            var name = user.Value<string>("displayName");
+            if (string.IsNullOrEmpty(name))
+            {
+                name = string.Format("{0} {1}", GetFirstName(user), GetLastName(user));
```

Comment:
Yea this is just ugly.
Will add the `GivenName` & `Surname` claims to `ClaimsIdentity` in the next commit.
btw, what is the value of having the `urn:microsoftaccount:*` claims?